src/chess/board.cc

```diff
@@ -969,21 +969,54 @@ MoveList ChessBoard::GenerateLegalMoves() const {
   return result;
 }
 
-void ChessBoard::SetFromFen(const std::string& fen, int* rule50_ply,
-                            int* moves) {
+void ChessBoard::SetFromFen(std::string fen, int* rule50_ply, int* moves) {
   Clear();
   int row = 7;
   int col = 0;
 
+  // Remove any trailing whitespaces to detect eof after the last field.
+  fen.erase(std::find_if(fen.rbegin(), fen.rend(),
+                         [](unsigned char c) { return !std::isspace(c); })
+                .base(),
+            fen.end());
+
   std::istringstream fen_str(fen);
   string board;
   string who_to_move;
   string castlings;
   string en_passant;
   int rule50_halfmoves;
   int total_moves;
-  fen_str >> board >> who_to_move >> castlings >> en_passant >>
-      rule50_halfmoves >> total_moves;
+  fen_str >> board;
+  if (fen_str.eof()) {
+    who_to_move = "w";
```

Comment:
Maybe initialize those values to their defaults in lines 984-989, and here have:
```
if (!fen_str.eof()) fen_str >> castlings;
// and so on
```

Saves quite a lot of vertical space. :)

Reply:
I interlaced the variable declaration / initialization with the conditional extraction to keep the castling comment closer.